Engine/Rendering/Viewport.cpp

```diff
@@ -0,0 +1,536 @@
+#include "Viewport.h"
+
+#include "Component/ComponentCamera.h"
+#include "Component/ComponentMeshRenderer.h"
+
+#include "Main/Application.h"
+#include "Main/GameObject.h"
+#include "Module/ModuleCamera.h"
+#include "Module/ModuleDebug.h"
+#include "Module/ModuleDebugDraw.h"
+#include "Module/ModuleEffects.h"
+#include "Module/ModuleEditor.h"
+#include "Module/ModuleProgram.h"
+#include "Module/ModuleRender.h"
+#include "Module/ModuleUI.h"
+#include "Module/ModuleSpacePartitioning.h"
+#include "Helper/Utils.h"
+
+#include "FrameBuffer/DepthFrameBuffer.h"
+#include "FrameBuffer/FrameBuffer.h"
+#include "LightFrustum.h"
+
+Viewport::Viewport(int options) : viewport_options(options)
+{
+	scene_quad = new Quad(2.f);
+
+	framebuffers.emplace_back(scene_fbo = new FrameBuffer(3));
+	framebuffers.emplace_back(ping_fbo = new FrameBuffer());
+	framebuffers.emplace_back(pong_fbo = new FrameBuffer());
+	framebuffers.emplace_back(postprocess_fbo = new FrameBuffer());
+	framebuffers.emplace_back(blit_fbo = new FrameBuffer());
+
+	depth_full_fbo = new DepthFrameBuffer();
+	depth_near_fbo = new DepthFrameBuffer();
+	depth_mid_fbo = new DepthFrameBuffer();
+	depth_far_fbo = new DepthFrameBuffer();
+
+	SetOutput(ViewportOutput::COLOR);
+}
+
+Viewport::~Viewport()
+{
+	delete scene_quad;
+
+	for (auto& framebuffer : framebuffers)
+	{
+		delete framebuffer;
+	}
+
+	delete depth_full_fbo;
+	delete depth_near_fbo;
+	delete depth_mid_fbo;
+	delete depth_far_fbo;
+}
+
+void Viewport::Render(ComponentCamera* camera)
+{
+	this->camera = camera;
+	camera->SetAspectRatio(width / height);
```

Comment:
Are you using the camera from the class or the variable? can be a little confusing, maybe changing the name of the variable to another one